src/variables.ts

```diff
@@ -3,6 +3,23 @@ import type { SchemaLike } from './introspection';
 import type { DocumentNodeLike } from './parser';
 import type { obj } from './utils';
 
+type GetInputObjectType<InputField, Introspection extends SchemaLike> = InputField extends {
+  name: any;
+  type: any;
+}
+  ? InputField extends { defaultValue?: undefined | null; type: { kind: 'NON_NULL' } }
+    ? {
+        [Name in InputField['name']]: unwrapTypeRec<InputField['type'], Introspection, true>;
```

Comment:
As per the other comment, `true` here in `unwrapTypeRec` becoming `false` should fix the issue of non-null/null not mattering, as far as I remember.

Last review's comment suggested to remove `GetInputObjectType` entirely, to make sure we don't accidentally break mutually/infinitely recursive input types